spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfigurationTest.java

```diff
@@ -55,64 +64,78 @@ public static void setUpBeforeClass() {
 		assertThat(this.getContext().containsBean("configurationChangeDetector")).isFalse();
 		assertThat(this.getContext().containsBean("configurationUpdateStrategy")).isFalse();
 	}
 
 	@Test
-	public void kubernetesConfigReloadWhenKubernetesDisabled() throws Exception {
-		setup("spring.cloud.kubernetes.enabled=false");
+	public void kubernetesConfigReloadWhenKubernetesDisabled() {
+		setup(KubernetesClientTestConfiguration.class, "spring.cloud.kubernetes.enabled=false");
 		assertThat(this.getContext().containsBean("configurationChangeDetector")).isFalse();
 		assertThat(this.getContext().containsBean("configurationUpdateStrategy")).isFalse();
 	}
 
 	@Test
-	public void kubernetesReloadEnabled() throws Exception {
-		setup("spring.cloud.kubernetes.enabled=true", "spring.cloud.kubernetes.config.enabled=true",
-				"spring.cloud.kubernetes.secrets.enabled=true", "spring.cloud.kubernetes.reload.enabled=true");
+	public void kubernetesReloadEnabled() {
+		setup(KubernetesClientTestConfiguration.class, "spring.cloud.kubernetes.enabled=true",
+				"spring.cloud.kubernetes.config.enabled=true", "spring.cloud.kubernetes.secrets.enabled=true",
+				"spring.cloud.kubernetes.reload.enabled=true");
 		assertThat(this.getContext().containsBean("configMapPropertySourceLocator")).isTrue();
 		assertThat(this.getContext().containsBean("secretsPropertySourceLocator")).isTrue();
 		assertThat(this.getContext().containsBean("configMapPropertyChangeEventWatcher")).isTrue();
 		assertThat(this.getContext().containsBean("secretsPropertyChangeEventWatcher")).isTrue();
 	}
 
 	@Test
-	public void kubernetesReloadEnabledButSecretDisabled() throws Exception {
-		setup("spring.cloud.kubernetes.enabled=true", "spring.cloud.kubernetes.config.enabled=true",
-				"spring.cloud.kubernetes.secrets.enabled=false", "spring.cloud.kubernetes.reload.enabled=true");
+	public void kubernetesReloadEnabledButSecretDisabled() {
+		setup(KubernetesClientTestConfiguration.class, "spring.cloud.kubernetes.enabled=true",
+				"spring.cloud.kubernetes.config.enabled=true", "spring.cloud.kubernetes.secrets.enabled=false",
+				"spring.cloud.kubernetes.reload.enabled=true");
 		assertThat(this.getContext().containsBean("configMapPropertySourceLocator")).isTrue();
 		assertThat(this.getContext().containsBean("secretsPropertySourceLocator")).isFalse();
 		assertThat(this.getContext().containsBean("configMapPropertyChangeEventWatcher")).isTrue();
 		assertThat(this.getContext().containsBean("secretsPropertyChangeEventWatcher")).isFalse();
 	}
 
 	@Test
-	public void kubernetesReloadEnabledButSecretAndConfigDisabled() throws Exception {
-		setup("spring.cloud.kubernetes.enabled=true", "spring.cloud.kubernetes.config.enabled=false",
-				"spring.cloud.kubernetes.secrets.enabled=false", "spring.cloud.kubernetes.reload.enabled=true");
+	public void kubernetesReloadEnabledButSecretAndConfigDisabled() {
+		setup(KubernetesClientTestConfiguration.class, "spring.cloud.kubernetes.enabled=true",
```

Comment:
`setup` method was slightly changed to get `Class<?>` as input, and I pass the `KubernetesClientTestConfiguration.class` configuration in there. everything else stays the same